features/dd-sdk-android-ndk/src/main/cpp/utils/backtrace-handler.cpp

```diff
@@ -45,53 +45,61 @@ namespace {
         return state.current - buffer;
     }
 
-    std::string address_to_hexa(uintptr_t address) {
+    void append_hex_address(uintptr_t address, std::string& string) {
         char address_as_hexa[20];
         // The ARM_32 processors will use an unsigned long long to represent a pointer so we will choose the
         // String format that fits both ARM_32 and ARM_64 (lx).
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wformat"
-        std::snprintf(address_as_hexa, sizeof(address_as_hexa), "0x%lx", address);
+        std::snprintf(address_as_hexa, sizeof(address_as_hexa), "0x%08lx", address);
 #pragma clang diagnostic pop
-        return std::string(address_as_hexa);
+        string.append(address_as_hexa);
+    }
+
+    void append_hex_address_no_prefix(uintptr_t address, std::string& string) {
+        char address_as_hexa[20];
+        // The ARM_32 processors will use an unsigned long long to represent a pointer so we will choose the
+        // String format that fits both ARM_32 and ARM_64 (lx).
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wformat"
+        std::snprintf(address_as_hexa, sizeof(address_as_hexa), "%08lx", address);
+#pragma clang diagnostic pop
+        string.append(address_as_hexa);
     }
 
     void get_info_from_address(size_t index,
                                const uintptr_t address,
-                               std::string *backtrace) {
+                               std::string& backtrace) {
         Dl_info info;
         int fetch_info_success = dladdr(reinterpret_cast<void *>(address), &info);
-        backtrace->append(std::to_string(index));
-        if (fetch_info_success) {
-
-            if (info.dli_fname) {
-                backtrace->append(" ");
-                backtrace->append(info.dli_fname);
-            }
+        backtrace.append(std::to_string(index));
```

Comment:
I am more afraid that this new format will not work with the current Android provided tools that can be used to de - symbolicate these traces:
https://docs.google.com/document/d/1eiszA5xT0LrVLpX_v2MHpB8oH_lg-Y7vdd3L4wBZHi0/edit#heading=h.jd2ss98jfyjo

Reply:
We should be okay, since the relative address we need to send to `addr2line` is still there, just moved to the front of the line rather than the back.

I'll update the documentation.